Tests/MongoSwiftTests/ServerSelectionTests.swift

```diff
@@ -5,132 +5,33 @@ import NIO
-    let standaloneServer = ServerDescription(type: .standalone)
-    let rsPrimaryServer = ServerDescription(type: .rsPrimary)
-    let rsSecondaryServer1 = ServerDescription(type: .rsSecondary, tags: ["dc": "ny", "rack": "2", "size": "large"])
-    let rsSecondaryServer2 = ServerDescription(type: .rsSecondary)
-    let rsSecondaryServer3 = ServerDescription(type: .rsSecondary, tags: ["dc": "ny", "rack": "3", "size": "small"])
-    let mongosServer = ServerDescription(type: .mongos)
-
-    // Read Preferences
-    let primaryReadPreference = ReadPreference(.primary)
-    let primaryPrefReadPreferemce = ReadPreference(.primaryPreferred)
-
-    // Tag Sets
-    let tagSet: BSONDocument = ["dc": "ny", "rack": "2"]
-    let tagSet2: BSONDocument = ["dc": "ny"]
-    let tagSet3: BSONDocument = ["size": "small"]
-
-    func testUnknownTopology() {
-        let unkownTopology = TopologyDescription(type: .unknown, servers: [standaloneServer])
-        expect(unkownTopology.findSuitableServers()).to(haveCount(0))
-    }
-
-    func testSingleTopology() {
-        let singleTopology = TopologyDescription(type: .single, servers: [standaloneServer])
-        expect(singleTopology.findSuitableServers()[0].type).to(equal(.standalone))
-    }
-
-    func testReplicaSetWithPrimaryTopology() {
-        let replicaSetTopology = TopologyDescription(type: .replicaSetWithPrimary, servers: [
-            rsPrimaryServer,
-            rsSecondaryServer1,
-            rsSecondaryServer2
-        ])
-        let replicaSetSuitableServers = replicaSetTopology
-            .findSuitableServers(readPreference: self.primaryReadPreference)
-        expect(replicaSetSuitableServers[0].type).to(equal(.rsPrimary))
-        expect(replicaSetSuitableServers).to(haveCount(1))
-
-        let replicaSetSuitableServers2 = replicaSetTopology
-            .findSuitableServers(readPreference: self.primaryPrefReadPreferemce)
-        expect(replicaSetSuitableServers2[0].type).to(equal(.rsPrimary))
-        expect(replicaSetSuitableServers2).to(haveCount(1))
-    }
-
-    func testReplicaSetNoPrimaryTopology() {
-        let replicaSetNoPrimaryTopology = TopologyDescription(type: .replicaSetNoPrimary, servers: [
-            rsSecondaryServer1,
-            rsSecondaryServer2
-        ])
-        let suitable1 = replicaSetNoPrimaryTopology
-            .findSuitableServers(readPreference: self.primaryReadPreference)
-        expect(suitable1).to(haveCount(0))
-
-        let suitable2 = replicaSetNoPrimaryTopology
-            .findSuitableServers(readPreference: nil)
-        expect(suitable2).to(haveCount(0))
-
-        let suitable3 = replicaSetNoPrimaryTopology
-            .findSuitableServers(readPreference: self.primaryPrefReadPreferemce)
-        expect(suitable3[0].type).to(equal(.rsSecondary))
-        expect(suitable3).to(haveCount(2))
+private struct ServerSelectionLogicTestFile: Decodable {
+    let topologyDescription: TopologyDescription
+    let operation: String
```

Comment:
nit: can we make this an enum with read/write cases? it's a little more code, but I think it makes the purpose of the property more obvious